### ISSUE TITLE
Implementing get_bgp_neighbors

### ISSUE THREAD

**Issue by aydineftekhari** [napalm-automation-community/napalm-cumulus#26]
I want to take a stab at implementing get_bgp_* for cumulus, starting with  get_bgp_neighbors
`net show bgp neighbor json` seems to have all (except *two of) the required info for `get_bgp_neighbors` (mapping bellow)

Can someone please bless this approach so I can get started on coding:

```cumulus@rtr1:~$ net show bgp neighbor json 
{
  "1.1.1.2":{
    "remoteAs":20,
    "localAs":10,
    "nbrExternalLink":true,
    "nbrDesc":"rtr2",
    "hostname":"rtr2",
    "bgpVersion":4,
    "remoteRouterId":"100.100.100.3",
    "bgpState":"Established",
    "bgpTimerUp":303067000,
    "bgpTimerUpMsec":303067000,
    "bgpTimerUpString":"3d12h11m",
    "bgpTimerUpEstablishedEpoch":1533761616,
    "bgpTimerLastRead":3000,
    "bgpTimerLastWrite":1000,
    "bgpInUpdateElapsedTimeMsecs":15287000,
    "bgpTimerHoldTimeMsecs":9000,
    "bgpTimerKeepAliveIntervalMsecs":3000,
    "neighborCapabilities":{
      "4byteAs":"advertisedAndReceived",
      "addPath":{
        "IPv4 Unicast":{
          "rxAdvertisedAndReceived":true
        }
      },
      "routeRefresh":"advertisedAndReceivedOldNew",
      "multiprotocolExtensions":{
        "IPv4 Unicast":{
          "advertisedAndReceived":true
        }
      },
      "hostName":{
        "advHostName":"rtr1",
        "advDomainName":"n\/a",
        "rcvHostName":"rtr2",
        "rcvDomainName":"n\/a"
      },
      "gracefulRestart":"advertisedAndReceived",
      "gracefulRestartRemoteTimerMsecs":120000,
      "addressFamiliesByPeer":"none"
    },
    "gracefulRestartInfo":{
      "endOfRibSend":{
        "IPv4 Unicast":true
      },
      "endOfRibRecv":{
        "IPv4 Unicast":true
      }
    },
    "messageStats":{
      "depthInq":0,
      "depthOutq":0,
      "opensSent":2,
      "opensRecv":2,
      "notificationsSent":0,
      "notificationsRecv":2,
      "updatesSent":6,
      "updatesRecv":8,
      "keepalivesSent":100906,
      "keepalivesRecv":100906,
      "routeRefreshSent":1,
      "routeRefreshRecv":0,
      "capabilitySent":0,
      "capabilityRecv":0,
      "totalSent":100915,
      "totalRecv":100918
    },
    "minBtwnAdvertisementRunsTimerMsecs":0,
    "addressFamilyInfo":{
      "IPv4 Unicast":{
        "updateGroupId":2,
        "subGroupId":2,
        "packetQueueLength":0,
        "inboundSoftConfigPermit":true,
        "commAttriSentToNbr":"extendedAndStandard",
        "acceptedPrefixCounter":1
      }
    },
    "connectionsEstablished":2,
    "connectionsDropped":1,
    "lastResetTimerMsecs":43869000,
    "lastResetDueTo":"BGP Notification received",
    "lastErrorCodeSubcode":"0606",
    "lastNotificationReason":"Cease\/Other Configuration Change",
    "hostLocal":"1.1.1.1",
    "portLocal":55880,
    "hostForeign":"1.1.1.2",
    "portForeign":179,
    "nexthop":"1.1.1.1",
    "nexthopGlobal":"fe80::a00:27ff:fee6:bfb2",
    "nexthopLocal":"fe80::a00:27ff:fee6:bfb2",
    "bgpConnection":"sharedNetwork",
    "connectRetryTimer":10,
    "estimatedRttInMsecs":6,
    "readThread":"on",
    "writeThread":"on"
  }
}
cumulus@rtr1:~$ 
```
 

NAPALM get_bgp_neighbors key | Cumulus NCLU key
-- | --
router_id | hostLocal
peers | nbrExternalLink
is_enabled': True | bgpState
address_family | bgpVersion
sent_prefixes | see note below
accepted_prefixes | acceptedPrefixCounter
received_prefixes | see note below
uptime | bgpTimerUpString
remote_as | remoteAs
is_up | bgpState
remote_id | hostForeign
local_as | localAs
description | nbrDesc

Note: sent_prefixes and received_prefixes counters are not readily available in `net show bgp neighbor json` we could count the number of prefixes by issuing the following commands for each neighbor: 

```cumulus@rtr1:~$ net show bgp neighbor 1.1.1.2 advertised-routes 
BGP table version is 4, local router ID is 100.100.100.2
Status codes: s suppressed, d damped, h history, * valid, > best, = multipath,
              i internal, r RIB-failure, S Stale, R Removed
Origin codes: i - IGP, e - EGP, ? - incomplete

   Network          Next Hop            Metric LocPrf Weight Path
*> 10.10.10.0/24    0.0.0.0                  0         32768 i
*> 20.20.20.0/24    1.1.1.2                                0 20 i

Total number of prefixes 2

cumulus@rtr1:~$ net show bgp neighbor 1.1.1.2 received-routes 
BGP table version is 0, local router ID is 100.100.100.2
Status codes: s suppressed, d damped, h history, * valid, > best, = multipath,
              i internal, r RIB-failure, S Stale, R Removed
Origin codes: i - IGP, e - EGP, ? - incomplete

   Network          Next Hop            Metric LocPrf Weight Path
*> 10.10.10.0/24    1.1.1.1                                0 20 10 i
*> 20.20.20.0/24    1.1.1.2                  0             0 20 i

Total number of prefixes 2
```

**Comment by mirceaulinic** [comment]
I don't have experience on Cumulus, but this makes sense to me in principle. Please go ahead and implement this @aydineftekhari. Thanks!

**Comment by aydineftekhari** [comment]
implementing get_bgp_neighbors  https://github.com/napalm-automation-community/napalm-cumulus/pull/40

**Comment by jmrenshaw** [comment]
Closed with PR #40.
